profile.js

```diff
@@ -0,0 +1,62 @@
+const profileForm = document.getElementById("profile-form");
+const accountNameElem = document.getElementById("account-name");
+const accountNumberElem = document.getElementById("account-number");
+const changePinElem = document.getElementById("change-pin");
+const currentPinElem = document.getElementById("current-pin");
+const newPinElem = document.getElementById("new-pin");
+const confirmNewPinElem = document.getElementById("confirm-new-pin");
+const closeBtnElem = document.getElementById("close-btn");
+const updateBtnElem = document.getElementById("update-btn");
+// display of aaccount number
+accountNumberElem.value = currentUserAccountNumber;
+
+const allUsers = JSON.parse(localStorage.getItem("MB_USER_ACCOUNTS"));
+
+function displayFormToUpdate() {
+  const pinUpdateSection = document.getElementById("pin-update-section");
+
+  if ((pinUpdateSection.style.display = "none")) {
+    pinUpdateSection.style.display = "block";
+  } else {
+    pinUpdateSection.style.display = "none";
+  }
+}
+function updateForm() {
+  if (
+    currentPinElem.value !==
+    getUserByAccountNumber(currentUserAccountNumber).accountPin
+  ) {
+    alert("Incorrect Pin");
+    return;
+  }
+  if (newPinElem.value !== confirmNewPinElem.value) {
+    alert("Pins do not match, try again");
+    return;
+  }
+
+  // finding the current user to update profile
+
+  let userToBeUpdated = allUsers.find(
+    (user) => user.accountNumber == accountNumberElem.value
+  );
+  userToBeUpdated.accountName = accountNameElem.value;
+  userToBeUpdated.accountPin = newPinElem.value;
+  // changing the updated properties to a string
+  const updatedObject = JSON.stringify(allUsers);
+  //  storing it back in local storage
+  localStorage.setItem("MB_USER_ACCOUNTS", updatedObject);
```

Comment:
You're not doing it right here. You haven't added the modified user to the users array. You're simply pushing back the same users array to local storage, which won't update the current user details.

See sample of how to update a user's details in deposit.js:
![Screenshot at Feb 22 9-00-21 PM](https://user-images.githubusercontent.com/51515797/220758216-541bbcb4-eb52-440a-9ffc-450eb7ed10f5.png)

See how it's done on line 31. 

Suggested pseudocode:
```
const currentUserIndex = getUserIndexByAccountNumber(currentUserAccountNumber);
allUsers[currentUserIndex].accountName = the updated account name;
// You need to check that user is updating PIN before you update it, otherwise, it will be updated to empty string
if (user is also changing PIN) {
  allUsers[currentUserIndex].accountPin = the updated account PIN;
}
setLocalStorageArrData("MB_USER_ACCOUNTS", allUsers);
```

Reply:
Done